src/Interpreters/HashJoin/HashJoin.h

```diff
@@ -433,8 +403,7 @@ class HashJoin : public IJoin
     /// Number of this flags equals to hashtable buffer size (plus one for zero value).
     /// Changes in hash table broke correspondence,
     /// so we must guarantee constantness of hash table during HashJoin lifetime (using method setLock)
-    mutable JoinStuff::JoinUsedFlags used_flags;
-
+    mutable std::unique_ptr<JoinStuff::JoinUsedFlags> used_flags;
```

Comment:
Why is it changed to  unique_ptr?

Reply:
to avoid include the header file which would cause nested including